src/test/ui/generator/yielding-in-match-guards.rs

```diff
@@ -0,0 +1,29 @@
+// check-pass
```

Comment:
This MVCE no longer ICEs for me, whereas this one does: https://github.com/rust-lang/rust/issues/72651#issuecomment-668720468. No need to leave credit, just an issue number is enough.

(Feel free to also include this one if you can confirm it did ICE at one point, though).

Reply:
The test case has been updated to refer to the said MVCE.

Reply:
@tmandry Forgot to mention this. This test case compiles if the async functions are not used in `fn main`. As you can see, as soon as async function `g` is called in `main`, broken MIR ICE can be reproduced. It only found this out recently.

Reply:
Ah, in that case let's put both test cases in!